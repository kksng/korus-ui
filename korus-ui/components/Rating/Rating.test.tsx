// @ts-nocheck
import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { Rating } from './index';

describe('Rating SNAPSHOTS', () => {
  it('should render', () => {
    const wrapper = mount(<Rating value={5} />);

    expect(wrapper.find('span.rating-item')).toHaveLength(5);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render controllable mode', () => {
    const wrapper = mount(<Rating value={5} />);

    expect(wrapper.find('span.rating-item')).toHaveLength(5);

    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.setProps({ value: 2 });

    expect(wrapper.find('span.rating-item.filled')).toHaveLength(2);

    expect(wrapper.find('span.rating-item:not(.filled)')).toHaveLength(3);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Rating HANDLERS', () => {
  it('should trigger onChange', () => {
    const onChange = jest.fn();

    const wrapper = mount(<Rating onChange={onChange} name="Ratty" value={2} />);

    wrapper
      .find('span')
      .first()
      .getDOMNode()
      .addEventListener('click', (ev) => wrapper.find('span').first().props().onClick(ev));

    wrapper.find('span.rating-item:not(.filled)').last().getDOMNode().dispatchEvent(new Event('click', { bubbles: true }));

    expect(onChange).toHaveBeenCalled();
  });
});

describe('Rating ATTRIBUTES', () => {
  it('should have value', () => {
    const wrapper = mount(<Rating value={4} />);

    expect(wrapper.find('span.rating-item.filled')).toHaveLength(4);

    expect(wrapper.find('span.rating-item:not(.filled)')).toHaveLength(1);
  });

  it('should have 5 stars by default', () => {
    const wrapper = mount(<Rating value={4} />);

    expect(wrapper.find('span.rating-item')).toHaveLength(5);
  });

  it('should accept max amount', () => {
    const wrapper = mount(<Rating value={4} max={10} />);

    expect(wrapper.find('span.rating-item')).toHaveLength(10);
  });
});
