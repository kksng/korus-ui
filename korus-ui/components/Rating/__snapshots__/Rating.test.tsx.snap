// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating SNAPSHOTS should render 1`] = `
<Rating
  value={5}
>
  <Span
    className="rating-wrapper"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <span
      className="rating-wrapper"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <Icon
        className="rating-item filled"
        key="0"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item filled"
        key="1"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item filled"
        key="2"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item filled"
        key="3"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item filled"
        key="4"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
    </span>
  </Span>
</Rating>
`;

exports[`Rating SNAPSHOTS should render controllable mode 1`] = `
<Rating
  value={5}
>
  <Span
    className="rating-wrapper"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <span
      className="rating-wrapper"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <Icon
        className="rating-item filled"
        key="0"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item filled"
        key="1"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item filled"
        key="2"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item filled"
        key="3"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item filled"
        key="4"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
    </span>
  </Span>
</Rating>
`;

exports[`Rating SNAPSHOTS should render controllable mode 2`] = `
<Rating
  value={2}
>
  <Span
    className="rating-wrapper"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <span
      className="rating-wrapper"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <Icon
        className="rating-item filled"
        key="0"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item filled"
        key="1"
      >
        <Span
          className="rating-item filled"
        >
          <span
            className="rating-item filled"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item"
        key="2"
      >
        <Span
          className="rating-item"
        >
          <span
            className="rating-item"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item"
        key="3"
      >
        <Span
          className="rating-item"
        >
          <span
            className="rating-item"
          />
        </Span>
      </Icon>
      <Icon
        className="rating-item"
        key="4"
      >
        <Span
          className="rating-item"
        >
          <span
            className="rating-item"
          />
        </Span>
      </Icon>
    </span>
  </Span>
</Rating>
`;
